draft-ietf-quic-transport.md

```diff
@@ -2256,130 +2297,144 @@ the server.
 +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
-|  [Gap N (8)]  |       [ACK Block N Length (8/16/32/64)]     ...
+                               ...
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+|                             Gap (i)                         ...
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+|                    Additional ACK Block (i)                 ...
 +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
 ~~~
 {: #ack-block-format title="ACK Block Section"}
 
-The fields in the ACK Block Section are:
+Each ACK Block acknowledges a contiguous range of packets by indicating the
+number of acknowledged packets that precede the largest packet number in that
+block.  A value of zero indicates that only the largest packet number is
+acknowledged.  Larger ACK Block values indicate a larger range, with
+corresponding lower values for the smallest packet number in the range.  Thus,
+given a largest packet number for the ACK, the smallest value is determined by
+the formula:
+
+~~~
+   smallest = largest - ack_block
+~~~
+
+The range of packets that are acknowledged by the ACK block include the range
+from the smallest packet number to the largest, inclusive.
+
+The largest value for the First ACK Block is determined by the Largest
+Acknowledged field; the largest for Additional ACK Blocks is determined by
+cumulatively subtracting the size of all preceding ACK Blocks and Gaps.
 
-First ACK Block Length:
+Each Gap indicates a range of packets that are not being acknowledged.  The
+number of packets in the gap is one higher than the encoded value of the Gap
+Field.
 
-: An unsigned packet number delta that indicates the number of contiguous
-  additional packets being acknowledged starting at the Largest Acknowledged.
+The value of the Gap field establishes the largest packet number value for the
+ACK block that follows the gap using the following formula:
+
+~~~
+  largest = previous_smallest - gap - 2
+~~~
 
-Gap To Next Block (opt, repeated):
+If the calculated value for largest or smallest packet number for any ACK Block
+is negative, an endpoint MUST generate a connection error of type FRAME_ERROR
+indicating an error in an ACK frame (that is, 0x10d).
 
-: An unsigned number specifying the number of contiguous missing packets from
-  the end of the previous ACK block to the start of the next.  Repeated "Num
-  Blocks" times.
+The fields in the ACK Block Section are:
 
-ACK Block Length (opt, repeated):
+First ACK Block:
 
-: An unsigned packet number delta that indicates the number of contiguous
-  packets being acknowledged starting after the end of the previous gap.
-  Repeated "Num Blocks" times.
+: A variable-length integer indicating the number of contiguous packets
+  preceding the Largest Acknowledged that are being acknowledged.
 
+Gap (repeated):
 
-#### Time Format
+: A variable-length integer indicating specifying the number of contiguous
```

Comment:
indicating specifying is redundant.  Also, you alternate between using indicating and specifying in these sections.  Is it worth consistently using one or the other?